cli/src/commands/extensions/api.rs

```diff
@@ -465,123 +405,83 @@ fn run_sandboxed(op_state: Rc<RefCell<OpState>>, process: Process) -> Result<Pro
-        },
+    // Add sandboxed command arguments.
+    sandbox_args.push("--".into());
+    sandbox_args.push(cmd.into());
+    for arg in &args {
+        sandbox_args.push(arg.into());
     }
-}
 
-/// Lock down the current process.
-#[cfg(unix)]
-fn lock_process(exceptions: permissions::Permissions, strict: bool) -> Result<()> {
-    let home_dir = dirs::home_dir()?;
+    // Execute sandboxed command.
+    let output = Command::new(env::current_exe()?)
+        .args(sandbox_args.iter_mut().map(|arg| arg.to_mut()))
+        .stdin(stdin)
+        .stdout(stdout)
+        .stderr(stderr)
+        .output()?;
+
+    Ok(ProcessOutput {
+        stdout: String::from_utf8_lossy(&output.stdout).into_owned(),
+        stderr: String::from_utf8_lossy(&output.stderr).into_owned(),
+        success: output.status.success(),
+        signal: output.status.signal(),
+        code: output.status.code(),
+    })
+}
 
-    let mut birdcage = if strict { Birdcage::new()? } else { permissions::default_sandbox()? };
+/// Convert [permissions::Permissions] to arguments for `phylum sandbox`.
+fn add_permission_args<'a>(
+    sandbox_args: &mut Vec<Cow<'a, str>>,
+    permissions: &'a permissions::Permissions,
+    strict: bool,
+) -> Result<()> {
+    if strict {
+        sandbox_args.push("--strict".into());
+    }
 
-    // Apply filesystem exceptions.
-    for path in exceptions.read.sandbox_paths().iter() {
+    // Add filesystem exception arguments.
+    let home_dir = dirs::home_dir()?;
+    for path in permissions.read.sandbox_paths().iter() {
         let path = dirs::expand_home_path(path, &home_dir);
-        permissions::add_exception(&mut birdcage, Exception::Read(path))?;
+        sandbox_args.push("--allow-read".into());
+        sandbox_args.push(path.to_string_lossy().into_owned().into());
     }
-    for path in exceptions.write.sandbox_paths().iter() {
+    for path in permissions.write.sandbox_paths().iter() {
         let path = dirs::expand_home_path(path, &home_dir);
-        permissions::add_exception(&mut birdcage, Exception::Write(path))?;
+        sandbox_args.push("--allow-write".into());
+        sandbox_args.push(path.to_string_lossy().into_owned().into());
     }
-    for path in exceptions.run.sandbox_paths().iter() {
+    for path in permissions.run.sandbox_paths().iter() {
         let path = dirs::expand_home_path(path, &home_dir);
-        let absolute_path = permissions::resolve_bin_path(path);
-        permissions::add_exception(&mut birdcage, Exception::ExecuteAndRead(absolute_path))?;
+        sandbox_args.push("--allow-run".into());
+        sandbox_args.push(path.to_string_lossy().into_owned().into());
     }
 
-    // Apply network exceptions.
-    if let permissions::Permission::Boolean(true) = exceptions.net {
-        birdcage.add_exception(Exception::Networking)?;
+    // Add network exception argument.
+    if let permissions::Permission::Boolean(true) = permissions.net {
```

Comment:
Should this be `if matches!(...)` instead of `if let`?

Reply:
I'd avoid `matches!` unless necessary tbh.

Reply:
More specifically by "necessary" I mean if you need to evaluate a `let` thing to true/false **and** chain something after it.